File: 이것이 취업을 위한 코딩 테스트다/구현/문자열_압축/문자열_압축.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = s.size();
    int max_offset = answer / 2 + 1;

    for (int offset = 1; offset < max_offset; offset += 1) {
      string current_word = s.substr(0, offset);
      string result = "";
      int count = 1;
      
      for (int i = offset; i < s.size(); i += offset) {
        string next_word = s.substr(i, offset);

        if (current_word == next_word) {
          count += 1;
        } else {
          result += (count > 1) ? (to_string(count)) + current_word : current_word;
          count = 1;
          current_word = next_word;
        }

      }
      result += (count > 1) ? (to_string(count)) + current_word: current_word;

      answer = min(int(result.size()), answer);
    }

    return answer;
}

File: 이것이 취업을 위한 코딩 테스트다/구현/문자열_압축/문자열_압축.java
class Solution {
  public int solution(String s) {
    int answer = s.length();
    int max_offset = answer / 2 + 1;

    for (int offset = 1; offset < max_offset; offset += 1) {
      String current_word = s.substring(0, offset);
      String result = "";
      int count = 1;

      for (int i = offset; i < s.length(); i += offset) {
        String next_word = "";
        for (int next_word_i = i; next_word_i < i + offset; next_word_i += 1) {
          if (next_word_i < s.length()) {
            next_word += s.charAt(next_word_i);
          }
        }

        if (current_word.equals(next_word)) {
          count += 1;
        } else {
          result += count > 1 ? (count + current_word) : current_word;
          count = 1;
          current_word = next_word;
        }
      }

      result += count > 1 ? (count + current_word) : current_word;
      answer = Math.min(answer, result.length());
    }

    return answer;
  }
}